crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -254,6 +254,19 @@ def prepare(options = {})
 
       protected
 
+      def finalize_cluster_upgrade(cluster)
+        save_upgrade_state("Finalizing the upgrade of nodes in cluster #{cluster}")
+
+        NodeObject.find("pacemaker_config_environment:#{cluster}").each do |node|
+          if node["pacemaker"]["founder"] && node.roles.include?("neutron-network")
+            node.run_ssh_cmd("source /root/.openrc; neutron-ha-tool --l3-agent-rebalance")
+          end
+          node["crowbar_wall"]["crowbar_upgrade_step"] = "done"
+          node.save
+        end
+        # FIXME: we need to enable chef-client and crowbar-join services at some point
```

Comment:
Hm.... that we could, true. Although, `join_and_chef` is still for one node only, executed when others are not upgraded.